DFC.ServiceTaxonomy.ApiFunction/Function/Execute.cs

```diff
@@ -0,0 +1,166 @@
+        private readonly IHttpRequestHelper _httpRequestHelper;
+        private readonly IJsonHelper _jsonHelper;
+        private readonly INeo4JHelper _neo4JHelper;
+        private readonly IFileHelper _fileHelper;
+
+        public Execute(IOptions<ServiceTaxonomyApiSettings> serviceTaxonomyApiSettings, IHttpRequestHelper httpRequestHelper, IJsonHelper jsonHelper, INeo4JHelper neo4JHelper, IFileHelper fileHelper)
+        {
+            _serviceTaxonomyApiSettings = serviceTaxonomyApiSettings ?? 
+                                          throw new ArgumentNullException(nameof(serviceTaxonomyApiSettings));
+
+            _httpRequestHelper = httpRequestHelper ??
+                                 throw new ArgumentNullException(nameof(httpRequestHelper));
+
+            _jsonHelper = jsonHelper ??
+                  throw new ArgumentNullException(nameof(jsonHelper));
+
+            _neo4JHelper = neo4JHelper ?? 
+                           throw new ArgumentNullException(nameof(neo4JHelper));
+
+            _fileHelper = fileHelper ??
+                          throw new ArgumentNullException(nameof(fileHelper));
+        }
+
+        [FunctionName("Execute")]
+        public async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = null)] HttpRequest req,
+            ILogger log)
+        {
+            var functionToProcess = _serviceTaxonomyApiSettings.Value.Function;
+
+            if(string.IsNullOrWhiteSpace(functionToProcess))
+                return new BadRequestObjectResult("Function cannot be found in app settings");
+
+            log.LogInformation(string.Format("{0} HTTP trigger function processed a request.", functionToProcess));
+
+            log.LogInformation("Attempting to read body from http request");
+
+            string requestBody = null;
+
+            try
+            {
+                requestBody = await _httpRequestHelper.GetBodyFromHttpRequest(req);
+            }
+            catch (IOException ex)
+            {
+                log.LogError("Unable to read body from req", ex);
+                return new BadRequestObjectResult("Unable to read body from req");
+            }
+
+            log.LogInformation("Attempting to Deserialize request body");
+
+            dynamic jsonBody;
+
+            try
+            {
+                jsonBody = JsonConvert.DeserializeObject(requestBody);
+            }
+            catch (JsonException ex)
+            {
+                log.LogError("Unable to retrieve body from req", ex);
+                return new UnprocessableEntityObjectResult("Unable to deserialize request body");
+            }
+
+            log.LogInformation("generating file name and dir to read json config");
+
+            var queryFileNameAndDir = string.Format(@"{0}\{1}.{2}", "CypherQueries" , functionToProcess, "json");
+
+            string cypherQueryJsonConfig = string.Empty;
```

Comment:
No need to initialise the string with a value, we could let it auto-initialise to null instead.
```suggestion
            string cypherQueryJsonConfig;
```